go-controller/pkg/ovn/address_set.go

```diff
@@ -418,3 +420,48 @@ func getIPv4ASHashedName(name string) string {
 func getIPv6ASHashedName(name string) string {
 	return hashedAddressSet(name + ipv6AddressSetSuffix)
 }
+
+// This method checks if an address set exists in OVN NB or not. In some cases when a namespace
+// is created, there will be multiple callbacks to create addtess set and create/updates ACLs for
+// network polocies. To ensure address set is created under these circumstances, this method
+// will retry for 1 second to check if the address set exists.
+func addressSetExists(hashName string) (bool, error) {
+	err := utilwait.PollImmediate(100*time.Millisecond, 1*time.Second, func() (bool, error) {
+		uuid, stderr, err := util.RunOVNNbctl("--data=bare",
+			"--no-heading", "--columns=_uuid", "find", "address_set",
+			"name="+hashName)
+		if err != nil {
+			klog.Warningf("Find failed to get address set %s, stderr: %s (%v). Retrying.",
+				hashName, stderr, err)
+		}
+		return uuid != "", nil
+	},
+	)
+	if err != nil {
+		return false, err
+	}
+	return true, nil
+}
+
+// This method checks if the address sets for a namespace exists or not. This is to ensure
+// when ACLs are created through namespace create callbacks for policies, we need to make sure
+// address set is already created before we install ACLs.
+func addressSetExistsForNamespace(namespace string) bool {
```

Comment:
shouldn't you first check the namespace to see if it has the address set before you check OVN?

Reply:
Since the race condition reported was complaining that address set referred in the ACL is not present in OVN, I was directly checking in OVN before adding ACL. Should I first the namespace still?